client/python/openlineage/client/transport/composite.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+from functools import cached_property
+from typing import TYPE_CHECKING, Any
+
+import attr
+from openlineage.client.transport.transport import Config, Transport
+from openlineage.client.utils import get_only_specified_fields
+
+if TYPE_CHECKING:
+    from openlineage.client.client import Event
+
+log = logging.getLogger(__name__)
+
+
+@attr.s
+class CompositeConfig(Config):
+    transports: dict[str, dict[str, Any]] = attr.ib()
+    continue_on_failure: bool = attr.ib(default=True)
+
+    @classmethod
+    def from_dict(cls, params: dict[str, Any]) -> CompositeConfig:
+        if "transports" not in params:
+            msg = "composite `transports` not passed to CompositeConfig"
+            raise RuntimeError(msg)
+        return cls(**get_only_specified_fields(cls, params))
+
+
+class CompositeTransport(Transport):
+    kind = "composite"
+    config_class = CompositeConfig
+
+    def __init__(self, config: CompositeConfig) -> None:
+        self.config = config
+
+    @cached_property
+    def transports(self) -> dict[str, Transport]:
+        return self._create_transports()
+
+    def _create_transports(self) -> dict[str, Transport]:
+        from openlineage.client.transport import get_default_factory
+
+        transports = {}
+        for transport_name, transport_config in self.config.transports.items():
+            transports[transport_name] = get_default_factory().create(transport_config)
+        return transports
+
+    def emit(self, event: Event) -> None:
+        for transport_name, transport in self.transports.items():
+            try:
+                log.debug("Emitting event using %s", transport_name)
+                transport.emit(event)
+            except Exception as e:  # noqa: BLE001
+                msg = f"Transport {transport_name} failed to emit event: {e}"
+                if self.config.continue_on_failure:
+                    log.debug(msg)
```

Comment:
Imho, we may not want to raise an error but we probably want to see it in the logs, without debug logs turned on. Maybe we should provide some more details on debug level, but i would consider either also logging on a warning / exception level or replace the debug log entirely.

Reply:
Yeah, I'd add warning level. This - CompositeTransport - is a clear case where user had to configure something themselves to get that result.

Reply:
Changed to warning level.